drafting/P2968Rx_ignore.md

```diff
@@ -115,33 +95,107 @@ struct  { // expostion only
 
 # Impact on existing code
 
-Since `std::ignore` is already implemented in a suitable way in all major C++ standard libraries, there is no impact on existing code.
+All three major standard library vendors (libstdc++, Microsoft, libc++) implement `std::ignore` in roughly the same
+way, but there are some minor differences that could be ironed out by this proposal.
+
+```c++
+// libstdc++
+struct __ignore_t {
+  constexpr const __ignore_t&
+    operator=(const auto&) const
+      { return *this; }
+};
+inline constexpr __ignore_t ignore{};
+
+// Microsoft STL
+struct __ignore_t {
+  constexpr const __ignore_t&
+    operator=(const auto&) const noexcept
+      { return *this; }
+};
+inline constexpr __ignore_t ignore{};
+
+// libc++
+template<class T>
+struct __ignore_t {
+  constexpr const __ignore_t&
+    operator=(auto&&) const
+      { return *this; }
+};
+inline constexpr __ignore_t<unsigned char> ignore = __ignore_t<unsigned char>();
+```
 
-However, may be LEWG will decide on being more specific and less hand-wavy on the semantics of the type underlying `std::ignore` and even follow the suggestion to move its definition into another header (`<utility>` or `<ignore>`).
+The programmer can detect whether `ignore`'s type is a template. We suggest either mandating that `ignore`'s type not be a template,
+or finding a way to leave it unspecified.
 
-If LWG decides on using code for the specification, libraries might want to adjust their implementation accordingly (which I believe is not required).
+The difference between `operator=(const auto&)` and `operator=(auto&&)` is that the latter will bind to volatile
+bit-field glvalues, and the former will not. This is <a href="https://cplusplus.github.io/LWG/issue3978">[LWG3978]</a>.
+We suggest resolving LWG3978 by mandating that `std::ignore = vbf;` be ill-formed; that is, the signature of
+`ignore::operator=` should take `(const auto&)` instead of `(auto&&)`.
```

Comment:
Oops! https://godbolt.org/z/14Yj7qWPz
`operator=(const auto&)` allows ordinary bitfields but not volatile ones.`operator=(auto&&)` does not allow any kind of bitfield — volatile or non-volatile.
So this should instead say,
"the _former_ will bind to _non_-volatile bit-field glvalues, and the _latter_ will not."
I think my conclusion remains the same — that we should mandate `(const auto&)` — but I did indeed screw up this entire paragraph. I'll repush in a moment.